pkg/volume/csi/csi_attacher.go

```diff
@@ -169,9 +169,13 @@ func (c *csiAttacher) waitForVolumeAttachmentInternal(volumeHandle, attachID str
 	if err != nil {
 		return "", fmt.Errorf("watch error:%v for volume %v", err, volumeHandle)
 	}
-
+	var watcherClosed bool
 	ch := watcher.ResultChan()
-	defer watcher.Stop()
+	defer func(closed bool) {
+		if !closed {
+			watcher.Stop()
+		}
+	}(watcherClosed)
```

Comment:
go evaluates arguments of deferred function at the point it evaluates `defer` statement, i.e. it will always call `func()` defined above with `false`.

You should IMO use `watcherClosed` variable directly inside the func body to get the actual value of the variable.

Reply:
Hi, thanks for your review and clarification. 
It's my mistake, I have updated it now.